store/store.go

```diff
@@ -186,24 +183,7 @@ func (s *Store[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, erro
 	if head := s.contiguousHead.Load(); head != nil {
 		return *head, nil
 	}
-
-	head, err := s.GetByHeight(ctx, s.heightSub.Height())
-	if err == nil {
-		return head, nil
-	}
-
-	var zero H
-	head, err = s.readHead(ctx)
-	switch {
-	default:
-		return zero, err
-	case errors.Is(err, datastore.ErrNotFound), errors.Is(err, header.ErrNotFound):
-		return zero, header.ErrNoHead
-	case err == nil:
-		s.heightSub.SetHeight(head.Height())
-		log.Infow("loaded head", "height", head.Height(), "hash", head.Hash())
-		return head, nil
-	}
+	return s.GetByHeight(ctx, s.heightSub.Height())
```

Comment:
ErrNoHead instead

if head.Load == nil => heightSub.Height() == 0

Reply:
Done (`return zero, header.ErrStoreEmpty` as in `.Tail`)